openzaak/src/main/kotlin/com/ritense/openzaak/liquibase/changelog/ChangeLog20220415ZaakInstanceLinkSetZaakTypeUrl.kt

```diff
@@ -0,0 +1,95 @@
+package com.ritense.openzaak.liquibase.changelog
+
+import com.ritense.valtimo.contract.authentication.AuthoritiesConstants.ADMIN
+import io.jsonwebtoken.Jwts
+import io.jsonwebtoken.SignatureAlgorithm
+import io.jsonwebtoken.security.Keys
+import liquibase.change.custom.CustomTaskChange
+import liquibase.database.Database
+import liquibase.database.jvm.JdbcConnection
+import liquibase.exception.ValidationErrors
+import liquibase.resource.ResourceAccessor
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.bodyToMono
+import java.nio.charset.Charset
+import java.util.Date
+
+class ChangeLog20220415ZaakInstanceLinkSetZaakTypeUrl : CustomTaskChange {
+
+    override fun execute(database: Database?) {
+        val connection = database!!.connection as JdbcConnection
+
+        val zaakToken = getOpenZaakToken(connection)
+        val zaakWebClient = getOpenZaakWebClient(zaakToken)
+        var statement = connection.prepareStatement("SELECT zaak_type_url, zaak_instance_url FROM zaak_instance_link")
+        val result = statement.executeQuery()
+
+        while (result.next()) {
+            val zaakTypeUrl = result.getString(1)
+            if (zaakTypeUrl == null) {
```

Comment:
wouldn't it be easier to filter this in the query?

Reply:
Changed